### PR TITLE
#6: Setup tests for module account-inbox

### LINKED ISSUE #6: Setup tests
The aim is to make unit tests runnable

Topic: https://github.com/OpenPaaS-Suite/esn/issues/11

Sample PR: https://github.com/OpenPaaS-Suite/esn-frontend-calendar/pull/25

### PR DESCRIPTION
**Pull request by billyfrost418**
setupT

### CODE REVIEW

**Comment by tuanlc** [comment]
The first commit is very weird??

File: package.json [hunk 1]
```diff
@@ -4,13 +4,53 @@
     "description": "Intermediary frontend module between Inbox and Account frontends for the OpenPaaS Suite - https://open-paas.org/",
     "main": "index.js",
     "scripts": {
-        "test": "echo \"Error: no test specified\" && exit 1",
-        "build:dev": "webpack"
+        "test": "./node_modules/karma/bin/karma start test/config/karma.conf.js",
+        "build:dev": "webpack",
+        "build:prod": "webpack --env=prod",
```
Comment (tuanlc) [inline package.json]: Why do you want to bundle this module?

File: package.json [hunk 2]
```diff
@@ -4,13 +4,53 @@
...
+        "angular-mocks": "1.3.20",
+        "babel-loader": "^8.1.0",
+        "babel-plugin-angularjs-annotate": "^0.10.0",
+        "chai": "3.0.0",
+        "chai-datetime": "1.4.0",
+        "chai-shallow-deep-equal": "1.4.0",
+        "clean-webpack-plugin": "^3.0.0",
+        "css-loader": "^3.6.0",
+        "dotenv-webpack": "^2.0.0",
+        "esn-frontend-login": "github:openpaas-suite/esn-frontend-login#main",
```
Comment (piavgh) [inline package.json]: I guess this login module is not needed anymore

File: src/app/components/identity/create/identity-create.controller.spec.js [hunk 3]
```diff
@@ -13,32 +13,32 @@ describe('The inboxIdentityCreateController', function() {
   beforeEach(function() {
     userId = 'userId';
 
-    module('esn.account-inbox');
+    angular.mock.module('esn.account-inbox')
```
Comment (piavgh) [inline src/app/components/identity/create/identity-create.controller.spec.js]: `;` is missing

File: test/config/mocks/modules.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (tuanlc) [inline test/config/mocks/modules.js]: remove this blank line

File: webpack.test.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  },
+  plugins: [
+    new webpack.NormalModuleReplacementPlugin(
+      /node_modules\/esn-frontend-common-libs\/src\/frontend\/js\/modules\/i18n\/i18n-loader.service.js/,
+      i18nLoaderMockPath
+    ),
+    new webpack.ProvidePlugin({
+      chai: chaiPath,
+      _: lodashPath,
+      moment: momentPath
```
Comment (tuanlc) [inline webpack.test.js]: Do we need moment for this module?

**Comment by billyfrost418** [comment]
That commit appears after I resolved a conflict right in the github page. However, I see no conflict between master branch and my branch when doing that. I will try to force push later to remove that commit

File: webpack.test.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    })
+  ],
+  module: {
+    rules: [
+      // We must ignore all *.run.js files
+      { test: /\.run.js$/,
+        loader: 'ignore-loader',
+        include: [
+          // path.resolve(__dirname, 'src/linagora.esn.unifiedinbox/app/module-registry.run.js'),
+          // path.resolve(__dirname, 'src/linagora.esn.unifiedinbox/app/search/search-query.run.js'),
```
Comment (tuanlc) [inline webpack.test.js]: ??

File: src/index.test.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+const shallowDeepEqual = require('chai-shallow-deep-equal/chai-shallow-deep-equal.js');
+const chaiDatetime = require('chai-datetime/chai-datetime.js');
+chai.use(sinonChai);
+chai.use(shallowDeepEqual);
+chai.use(chaiDatetime);
+
+// require all test files using special Webpack feature
+// https://webpack.github.io/docs/context.html#require-context
+const testsContext = require.context(".", true, /\.spec$/);
+testsContext.keys().forEach(testsContext);
```
Comment (piavgh) [inline src/index.test.js]: please add a new line at the end of file

File: test/config/karma.conf.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    preprocessors: {
+      'src/index.test.js': ['webpack']
+    },
+
+    webpack: webpackConfig,
+    plugins: [
+      'karma-firefox-launcher',
+      'karma-mocha',
+      'karma-webpack',
+      'karma-coverage',
```
Comment (piavgh) [inline test/config/karma.conf.js]: please remove coverage as in the PR of Huy

File: webpack.test.js (hunk 5 above)
Comment (billyfrost418) [reply webpack.test.js]: My bad behaviour of copying files. I have removed this

File: webpack.test.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    new webpack.ProvidePlugin({
+      chai: chaiPath,
+      _: lodashPath,
+    })
+  ],
+  module: {
+    rules: [
+      // We must ignore all *.run.js files
+      { test: /\.run.js$/,
+        loader: 'ignore-loader',
```
Comment (itswillta) [inline webpack.test.js]:
```suggestion
      { 
        test: /\.run.js$/,
        loader: 'ignore-loader',
```

File: webpack.test.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+      i18nLoaderMockPath
+    ),
+    new webpack.ProvidePlugin({
+      chai: chaiPath,
+      _: lodashPath,
+    })
+  ],
+  module: {
+    rules: [
+      // We must ignore all *.run.js files
```
Comment (itswillta) [inline webpack.test.js]: Delete this comment.

File: package.json [hunk 11]
```diff
@@ -4,13 +4,51 @@
     "description": "Intermediary frontend module between Inbox and Account frontends for the OpenPaaS Suite - https://open-paas.org/",
     "main": "index.js",
     "scripts": {
-        "test": "echo \"Error: no test specified\" && exit 1",
-        "build:dev": "webpack"
+        "test": "./node_modules/karma/bin/karma start test/config/karma.conf.js",
+        "build:dev": "webpack",
```
Comment (tuanlc) [inline package.json]: We don't need this one too

**Review by tuanlc — DISMISSED** [review]

File: package.json [hunk 12]
```diff
@@ -4,13 +4,51 @@
...
+        "chai-shallow-deep-equal": "1.4.0",
+        "clean-webpack-plugin": "^3.0.0",
+        "css-loader": "^3.6.0",
+        "dotenv-webpack": "^2.0.0",
+        "expose-loader": "^1.0.0",
+        "file-loader": "^6.0.0",
+        "html-loader": "^1.1.0",
+        "html-webpack-plugin": "^4.3.0",
+        "ignore-loader": "^0.1.2",
+        "karma-coverage": "1.1.2",
```
Comment (itswillta) [inline package.json]: Remove this.

File: package.json [hunk 13]
```diff
@@ -4,13 +4,51 @@
...
+        "pug-loader": "^2.4.0",
+        "sinon": "6.1.4",
+        "sinon-chai": "3.2.0",
+        "style-loader": "^1.2.1",
+        "svg-inline-loader": "^0.8.2",
+        "terser-webpack-plugin": "^3.0.6",
+        "url-loader": "^4.1.0",
+        "webpack": "^4.43.0",
+        "webpack-cli": "^3.3.11",
+        "webpack-dev-server": "^3.11.0",
```
Comment (itswillta) [inline package.json]: Remove this.

File: package.json [hunk 14]
```diff
@@ -4,13 +4,51 @@
...
         "esn-frontend-inbox": "github:OpenPaaS-Suite/esn-frontend-inbox#main"
+    },
+    "devDependencies": {
+        "angular-mocks": "1.3.20",
+        "babel-loader": "^8.1.0",
+        "babel-plugin-angularjs-annotate": "^0.10.0",
+        "chai": "3.0.0",
+        "chai-datetime": "1.4.0",
+        "chai-shallow-deep-equal": "1.4.0",
+        "clean-webpack-plugin": "^3.0.0",
```
Comment (itswillta) [inline package.json]: We don't need this.

File: package.json [hunk 15]
```diff
@@ -4,13 +4,51 @@
...
     "author": "",
     "license": "AGPLv3",
     "dependencies": {
         "esn-frontend-common-libs": "github:openpaas-suite/esn-frontend-common-libs#main",
         "esn-frontend-inbox": "github:OpenPaaS-Suite/esn-frontend-inbox#main"
+    },
+    "devDependencies": {
+        "angular-mocks": "1.3.20",
+        "babel-loader": "^8.1.0",
+        "babel-plugin-angularjs-annotate": "^0.10.0",
```
Comment (itswillta) [inline package.json]: We don't need these.

**Review by itswillta — COMMENTED** [review]
There are a lot of production depedencies which are redundant. Please check and remove them all.

File: package.json [hunk 16]
```diff
@@ -4,13 +4,47 @@
...
     },
     "author": "",
     "license": "AGPLv3",
     "dependencies": {
         "esn-frontend-common-libs": "github:openpaas-suite/esn-frontend-common-libs#main",
         "esn-frontend-inbox": "github:OpenPaaS-Suite/esn-frontend-inbox#main"
+    },
+    "devDependencies": {
+        "angular-mocks": "1.3.20",
+        "babel-loader": "^8.1.0",
```
Comment (itswillta) [inline package.json]: No need for this.

File: package.json [hunk 17]
```diff
@@ -4,13 +4,47 @@
...
+        "less-loader": "^6.1.1",
+        "mocha": "6.2.0",
+        "pug": "^3.0.0",
+        "pug-html-loader": "^1.1.5",
+        "pug-loader": "^2.4.0",
+        "sinon": "6.1.4",
+        "sinon-chai": "3.2.0",
+        "style-loader": "^1.2.1",
+        "svg-inline-loader": "^0.8.2",
+        "terser-webpack-plugin": "^3.0.6",
```
Comment (itswillta) [inline package.json]: No need for this.

File: test/config/karma.conf.js [hunk 18]
```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const webpackConfig = require('../../webpack.test');
+
+module.exports = function(config) {
+  const singleRun = process.env.SINGLE_RUN ? process.env.SINGLE_RUN !== 'false' : true;
```
Comment (itswillta) [inline test/config/karma.conf.js]:
```suggestion
  const singleRun = process.env.SINGLE_RUN !== 'false';
```

**Review by tuanlc — APPROVED** [review]

**Review by piavgh — APPROVED** [review]

**Review by itswillta — APPROVED** [review]

**Pull request merged**